src/lib/languages/mod.rs

```diff
@@ -0,0 +1,32 @@
+mod javascript;
+
+use super::*;
+pub use javascript::JavaScript;
+
+#[derive(Debug)]
+pub struct LangError {
+  language: String
+}
+
+pub enum Lang {
+  JS
+}
+
+impl Into<&'static str> for Lang {
```

Comment:
I think you want to do the other way around here we sure want to sometimes show the language it's name we're parsing but here we want to convert the user inputted language into Lang so:  
```rust 
impl Into<Lang> for String {

// OR
impl<'a> Into<Lang> for &'a str {

// OR (here you can input 'str or String either way it will be converted to String)
impl Into<Lang> for Into<String> { 
```

Reply:
I can't quite figure out how to finish this implementation. I was hoping you would. I got this far: https://github.com/talpalang/talpa/pull/4/commits/a6bc2efb8d03f034486ba56e97dde3c8989c87b0